docs/container-vm-image-wrapper.md

```diff
@@ -0,0 +1,238 @@
+# VM Images in Container Registery
+
+## Motivation
+
+KubeVirt is a drop in solution for running VMs on an existing kubernetes
+cluster. One of our goals in doing this is to leverage the tools and workflows
+kubernetes has established for containers and apply them to virtualized
+workloads.
+
+For example, with KubeVirt VMs are managed like pods. Users post VMs to
+kubernetes like pods and the VMs will soon have the option to integrate into
+the kubernetes overlay network like pods.
+
+To continue down the path of allowing users to introduce VM workloads into
+an existing kubernetes cluster with as little friction as possible, the next
+hurdle for us to overcome is tieing VM image distribution into the same
+mechanism used by pods. To do this, KubeVirt must have a way of storing and
+accessing VM images using a container registry.
+
+## Requirements
+* Users must have a workflow for pushing VM images into a container registry.
+* KubeVirt runtime must be able to consume VM images stored in the container
+  registry.
+
+## Initial Use Cases
+For the sake of simplicity, this proposal is focusing on the details involved
+with use cases where VMs are launched using ephemeral storage that is not
+persistent between VM restarts.
```

Comment:
I think it is actually helpful to consider persistent storage at the same time. Currently we've just been considering that a VM boots from a PVC without ever asking how you get content into these PVs in the first place. This could naturally be done by populating it from a master image at some point. Conversely if you have a PVC with some content in it, we shouldn't be restricted to only using it as a direct writable volume. It should be possible to use that PVC as the backing store for an ephemeral image. 

IOW, I think the persistent vs ephemeral disk distinction ought to be independent of the how we get content into the volumes that back the disk. So I'd not see use of docker being something that is directly associated with the VM startup process - a VM spec just refers to some generic storage location  (whether a PVC or an equivalent concept that might refer to a file on a filesystem). Docker integration is then just one (optional) way of getting data into these PVCs or files.